clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -899,6 +907,12 @@ public void unsubscribe() {
     @Override
     public void assign(Collection<TopicPartition> partitions) {
         acquire();
+
+        if (!this.assignment().isEmpty() && this.autoCommitEnabled)
```

Comment:
One other thing. Would it make sense to move this call to the `else` branch here (just before the call to `assignFromUser`)? Otherwise, we'd double-commit down the `unsubscribe` path.